static/app/components/nav/config.tsx

```diff
@@ -0,0 +1,192 @@
+import {useMemo} from 'react';
+
+import type {NavItemRaw, SidebarItem, SubmenuItem} from 'sentry/components/nav/utils';
+import {
+  ActiveStatus,
+  getActiveStatus,
+  NAV_DIVIDER,
+  resolveSidebarItem,
+  splitAtDivider,
+} from 'sentry/components/nav/utils';
+import {
+  IconDashboard,
+  IconGraph,
+  IconIssues,
+  IconLightning,
+  IconProject,
+  IconSearch,
+  IconSettings,
+  IconSiren,
+} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import {getDiscoverLandingUrl} from 'sentry/utils/discover/urls';
+import {useLocation} from 'sentry/utils/useLocation';
+import useOrganization from 'sentry/utils/useOrganization';
+import {useModuleURLBuilder} from 'sentry/views/insights/common/utils/useModuleURL';
+import {MODULE_SIDEBAR_TITLE} from 'sentry/views/insights/http/settings';
+import {MODULE_TITLES} from 'sentry/views/insights/settings';
+import {getSearchForIssueGroup, IssueGroup} from 'sentry/views/issueList/utils';
+
+export interface NavItemsResult {
+  primary: {
+    body: SidebarItem[];
+    footer: SidebarItem[];
+  };
+  secondary: {
+    body: SubmenuItem[];
+    footer: SubmenuItem[];
+  };
+}
```

Comment:
```suggestion
export interface NavItemsResult {
  primary: {
    body: ReadonlyArray<SidebarItem>;
    footer: ReadonlyArray<SidebarItem>;
  };
  secondary: {
    body: ReadonlyArray<SidebarItem>;
    footer: ReadonlyArray<SidebarItem>;
  };
}
```
I think you want to make sure these are not mutated in any component as it would impact the rendering